Test/Models/test_nn.py

```diff
@@ -22,26 +22,31 @@
 import pytest
 
 
-def _setup_symm(symmetries, N):
-    hi = nk.hilbert.Spin(1 / 2, N)
+def _setup_symm(symmetries, N, lattice="chain"):
+
+    if lattice == "chain":
+        g = nk.graph.Chain(N)
+    else:
+        g = nk.graph.Square(N)
```

Comment:
You could just pass a function, like
```suggestion
def _setup_symm(symmetries, N, lattice=nk.graph.Chain):
     g = lattice(N)
```
and change the calling sites accordingly. That saves the additional `if` blocks.

Reply:
this works for me